packages/passport/sdk/src/starkEx/passportImxProvider.ts

```diff
@@ -213,55 +217,71 @@ export class PassportImxProvider implements IMXProvider {
   }
 
   async createOrder(request: UnsignedOrderRequest): Promise<imx.CreateOrderResponse> {
-    const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
-
-    return createOrder({
-      request,
-      user,
-      starkSigner,
-      ordersApi: this.immutableXClient.ordersApi,
-      guardianClient: this.guardianClient,
-    });
+    return this.guardianClient.withDefaultConfirmationScreenTask(
+      async () => {
+        const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
+        return createOrder({
+          request,
+          user,
+          starkSigner,
+          ordersApi: this.immutableXClient.ordersApi,
+          guardianClient: this.guardianClient,
+        });
+      },
+    )();
   }
 
   async cancelOrder(
     request: imx.GetSignableCancelOrderRequest,
   ): Promise<imx.CancelOrderResponse> {
-    const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
-
-    return cancelOrder({
-      request,
-      user,
-      starkSigner,
-      ordersApi: this.immutableXClient.ordersApi,
-      guardianClient: this.guardianClient,
-    });
+    return this.guardianClient.withDefaultConfirmationScreenTask(
+      async () => {
+        const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
+
+        return cancelOrder({
+          request,
+          user,
+          starkSigner,
+          ordersApi: this.immutableXClient.ordersApi,
+          guardianClient: this.guardianClient,
+        });
+      },
+    )();
   }
 
   async createTrade(request: imx.GetSignableTradeRequest): Promise<imx.CreateTradeResponse> {
-    const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
-
-    return createTrade({
-      request,
-      user,
-      starkSigner,
-      tradesApi: this.immutableXClient.tradesApi,
-      guardianClient: this.guardianClient,
-    });
+    return this.guardianClient.withDefaultConfirmationScreenTask(
+      async () => {
+        const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
+
+        return createTrade({
+          request,
+          user,
+          starkSigner,
+          tradesApi: this.immutableXClient.tradesApi,
+          guardianClient: this.guardianClient,
+        });
+      },
+    )();
   }
 
   async batchNftTransfer(
     request: NftTransferDetails[],
   ): Promise<imx.CreateTransferResponse> {
-    const { user, starkSigner } = await this.#getRegisteredImxUserAndSigners();
-
-    return batchNftTransfer({
-      request,
-      user,
-      starkSigner,
-      transfersApi: this.immutableXClient.transfersApi,
-      guardianClient: this.guardianClient,
-    });
+  // eslint-disable-next-line function-paren-newline
```

Comment:
```suggestion
```
This doesn't seem necessary

Reply:
You're right - it was a holdover from when I moved it from the workflow file. I'll remove that